pandas/io/packers.py

```diff
@@ -371,9 +385,8 @@ def encode(obj):
             return {'typ': 'block_manager',
                     'klass': obj.__class__.__name__,
                     'axes': data.axes,
-                    'blocks': [{'items': data.items.take(b.mgr_locs),
-                                'locs': b.mgr_locs.as_array,
-                                'values': convert(b.values),
+                    'blocks': [{'locs': b.mgr_locs.as_array,
+                                'values': convert_if_not_dt_index(b.values),
```

Comment:
this is way hacky. you shouldn't need to specifically do this (test for ``DatetimeIndex``. It *should* just be converted normally. The entire difference here is the dispatch on the dtype, which in this case would be a string.